streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -124,6 +124,8 @@ public void configure(Map<String, ?> configs) {
             internalTopicManager = new InternalTopicManager(
                     (String) configs.get(StreamsConfig.ZOOKEEPER_CONNECT_CONFIG),
                     configs.containsKey(StreamsConfig.REPLICATION_FACTOR_CONFIG) ? (Integer) configs.get(StreamsConfig.REPLICATION_FACTOR_CONFIG) : 1);
+        } else {
+            log.info("Config '{}' isn't supplied and hence no internal topics will be created.", StreamsConfig.ZOOKEEPER_CONNECT_CONFIG);
```

Comment:
We are not logically defining ZOOKEEPER_CONNECT_CONFIG a required config in StreamsConfig, which we debated whether or not we need to. The reasons we decided to not make a required config is that it is actually not required for all applications: think of a piping application.